client/src/components/Student/DashboardDetails.tsx

```diff
@@ -51,6 +84,46 @@ export function DashboardDetails(props: Props) {
                   Issue Certificate
                 </Button>
               </Popconfirm>
+              {props.isAdmin && (
+                <>
+                  <Button
+                    danger
+                    icon={<FileExcelOutlined style={{ color: 'red' }} />}
+                    onClick={() => setModalOpen(true)}
+                    loading={props.isLoading}
+                  >
+                    Remove Certificate
+                  </Button>
+                  <Modal
+                    title="Confirm and remove the certificate"
+                    open={modalOpen}
+                    onOk={handleModalConfirm}
+                    onCancel={handleModalCancel}
+                    afterOpenChange={setModalInputFocus}
+                    width={350}
+                    okButtonProps={{
+                      disabled: inputValue !== details.githubId,
+                      danger: true,
+                    }}
+                    okText="Confirm"
+                    cancelText="Cancel"
+                    destroyOnClose
+                  >
+                    <div style={{ padding: '8px 0' }}>
+                      <p>
+                        Type <strong>{details.githubId}</strong> to confirm:
+                      </p>
+                      <Input
+                        ref={modalInputRef}
+                        placeholder="GitHub username"
+                        value={inputValue}
+                        onChange={e => setInputValue(e.target.value)}
+                        onPressEnter={handleModalConfirm}
+                      />
+                    </div>
+                  </Modal>
+                </>
+              )}
```

Comment:
Please extract it into separated component, it has too much logic to put it inline